packages/protocol/contracts/L1/TaikoL1.sol

```diff
@@ -162,9 +163,14 @@ contract TaikoL1 is EssentialContract, ITaikoL1, TaikoEvents {
 
         TaikoData.Config memory config = getConfig();
 
+        IVerifier.Context[] memory ctxs = new IVerifier.Context[](_blockIds.length);
+        TaikoData.Transition[] memory trans = new TaikoData.Transition[](_blockIds.length);
+
         for (uint256 i; i < _blockIds.length; ++i) {
-            _proveBlock(_blockIds[i], _inputArr[i], config);
+            (ctxs[i], trans[i]) = _proveBlock(_blockIds[i], _inputArr[i], config);
         }
+
+        IVerifier(verifier).verifyProof(ctxs, trans, proof);
```

Comment:
I have a compilation error, due to this outter `verifyProof()`, the `verifier` variable is not know. It is coming from the tier configuration itself (now, in our current design).

And also this yields into another problem, with an example:
The block tiers (and hence the tier verifiers) can be for example:
block nr. 102: SGX only
block nr. 103: SGX+RISC0
block nr. 104: SGX+SP1
block nr. 105: SGX only again

And thus, the verifier contract address itself differs from block to block. so aggregating multiple, continuous blocks will only work (in the current design at least!) if we have the exact same logic (contract) to verify (so same tiers), otherwise no.

Reply:
I think you can only aggregate proofs for the same tier together, but the proofs may correspond to blocks that are not consecutive.

Reply:
Yes, but then with ever 500-100th zk tier blocks in practice it will be useful for tee only. (At least for now.)

Reply:
I think I have now added the required checks to make sure only the same proofs can be aggregrated.